libp2p/pubsub/pubsub.py

```diff
@@ -336,6 +403,14 @@ def handle_subscription(self, origin_id: ID, sub_message: Any) -> None:
             return
 
         # TODO: - Validate the message. If failed, reject it.
+        # Validate the signature of the message
+        # FIXME: `signature_validator` is currently a stub.
+        if not signature_validator(msg.key, msg.SerializeToString()):
+            return
+        # Validate the message with registered topic validators
+        is_validation_passed = await self.validate_msg(msg_forwarder, msg)
+        if not is_validation_passed:
+            return
```

Comment:
This early return is a little confusing to me.  Reading the code (briefly) doesn't make it clear why we would return early on validation passing...  Maybe expand the code comment to try and explain this?